crates/vm/backends/mod.rs

```diff
@@ -335,7 +335,7 @@ impl Evm {
                 store_wrapper,
                 block_cache,
             } => {
-                store_wrapper.block_hash = header.parent_hash;
+                store_wrapper.block_hash = header.compute_block_hash();
```

Comment:
Although the fix is necessary, we should, in future PR, somehow cache the block hash to avoid calculate it on every simulate, as it may be too expensive.